dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/NettyChain.java

```diff
@@ -0,0 +1,393 @@
+                Tr.debug(tc, "Netty Chain is not in a stoppable state. Current state: " + state.get());
+            }
+        }
+
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.exit(this, tc, "stop chain " + this);
+        }
+    }
+
+    private void stopAndWait() {
+        if (state.get() != ChainState.STOPPED && state.get() != ChainState.UNINITIALIZED) {
+            stop();
+        }
+    }
+
+    @Override
+    public synchronized void update(String resolvedHostName) {
+
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.entry(this, tc, "Updating Netty Chain  " + endpointName + " Current state: " + state.get());
+        }
+
+        if (!enabled || FrameworkState.isStopping()) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.exit(this, tc, "Chain is disabled or framework is stopping, skipping update ");
+            }
+            return;
+        }
+
+        final ActiveConfiguration oldConfig = currentConfig;
+        final ActiveConfiguration newConfig = new ActiveConfiguration(isHttps(), getOwner().getTcpOptions(), isHttps() ? getOwner().getSslOptions() : null, getOwner().getHttpOptions(), getOwner().getRemoteIpConfig(), getOwner().getCompressionConfig(), getOwner().getSamesiteConfig(), getOwner().getHeadersConfig(), getOwner().getEndpointOptions(), resolvedHostName);
+
+        if (newConfig.configPort < 0 || !newConfig.complete()) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(this, tc, "Stopping chain due to configuration " + newConfig);
+            }
+            // save the new/changed configuration before we start setting up the new chain
+            currentConfig = newConfig;
+            stopAndWait();
+            state.set(ChainState.UNINITIALIZED);
+        }
+
+        else {
+
+            if (!newConfig.unchanged(oldConfig)) {
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(this, tc, "This configuration differs and should cause an update ");
+                }
+                currentConfig = newConfig;
+                if (state.get() != ChainState.UNINITIALIZED) {
+                    stopAndWait();
+                }
+            }
+            startNettyChannel();
+        }
+
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(this, tc, "Channel restarted with new configuration");
+        }
+    }
+
+    public synchronized void startNettyChannel() {
+
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.entry(this, tc, "Starting Netty Channel: " + endpointName + ", Current state: " + state.get() + ", Enabled: " + enabled);
+        }
+
+        // if (currentConfig == null || !currentConfig.complete() || !enabled || FrameworkState.isStopping()) {
```

Comment:
We should clean this up if not necessary

Reply:
I'll do this